package.json

```diff
@@ -150,6 +150,7 @@
     "eslint-plugin-promise": "6.2.0",
     "eslint-plugin-react": "7.34.2",
     "eslint-plugin-react-native": "4.1.0",
+    "ignore-styles": "^5.0.1",
```

Comment:
Is this import necessary? The repo has been archived as of 2 years ago.

Reply:
Hmm, I can suggest a slightly different approach: instead of exporting the whole `highlightNodeMatchingSearchTerm` function, separate its text search functionality into its own method and place it in `utils/source-parsing.js`. Something like this:
```js
export function findNodeMatchingSearchTerm(nodeText, searchText) {
  if (!searchText || !nodeText) {
    return null;
  }

  const index = nodeText.toLowerCase().indexOf(searchText.toLowerCase());
  if (index < 0) {
    return null;
  }
  const prefix = nodeText.substring(0, index);
  const suffix = nodeText.slice(index + searchText.length);
  // Matched word will be wrapped in a separate span for custom highlighting
  const matchedWord = nodeText.slice(index, index + searchText.length);
  return {prefix, matchedWord, suffix};
}
```

Then `highlightNodeMatchingSearchTerm` can be as follows:
```js
const highlightNodeMatchingSearchTerm = (nodeText, searchText) => {
  const searchResults = findNodeMatchingSearchTerm(nodeText, searchText);
  if (!searchResults) {
    return nodeText;
  }
  const {prefix, matchedWord, suffix} = searchResults;
  return (
    <>
      {prefix}
      <span className="search-word-highlighted" data-match={searchText}>
        {matchedWord}
      </span>
      {suffix}
    </>
  );
};
```

Then you can change the unit tests to validate `findNodeMatchingSearchTerm` (in `utils-source-parsing-specs.js`), and this extra import will not be necessary.

Reply:
Got it. i was banging my head around looking for ways to enable the testability. This make more sense and will make the necessary changes.